### ISSUE TITLE
Possible to run with separate settings for different files?

### ISSUE THREAD

**Issue by dan-lind** [eirnym/js2p-gradle#31]
More of a question than an actual issue: 

I was wondering if it is somehow possible to run with separate settings for different files? In my example, I have schemas that look a little bit different, but have overlapping names. As such, I need to prefix each of the generate pojos a bit differently. 

I have currenly resolved to using a bash script and the CLI like this

```
jsonschema2pojo -y Create -fdt -r -ut -tv 11 -303 -b -R -s src/main/resources/schemas/CREATE-Payload_Schema.json -t build/generated/sources/js2p -p com.example
jsonschema2pojo -y Update -fdt -r -ut -tv 11 -303 -b -s src/main/resources/schemas/UPDATE_Payload_Schema.json -t build/generated/sources/js2p -p com.example
jsonschema2pojo -y Read -fdt -r -ut -tv 11 -303 -b -s src/main/resources/schemas/READ_Response_Schema.json -t build/generated/sources/js2p -p com.example
```

This works but is giving us some headache as some developers or on a Windows machine.

**Comment by eirnym** [comment]
Hi @dan-lind 

Could you help me by providing an example, how do you see a configuration for such scenario in a single `build.gradle`? I'm really curious how to implement this correctly. 

Most plugins generating code I know prefer to have single configuration per build file. In a few projects I use a common include file `$projectDir/gradle/json.gradle` to provide most of settings for such modules and so module itself looks like. This is not an implementation of your wish.

```gradle
ext {
   setting1 = "value1"
   setting2 = "value2"
}
apply from: "$projectDir/gradle/json.gradle"
```

**Comment by dan-lind** [comment]
Yeah I've been thinking about it but not quite sure what would be the best way. 
I was thinking about something along the lines of an array of source files, where each files is represented by a map holding the path for that specific file plus any specific settings for that file. 

Something along those lines (not sure this is syntactically correct but I hope the idea comes across)

```
jsonSchema2Pojo {

  //Default settings to be applied to all files unless otherwise specified
  generateBuilders = true
  usePrimitives = true
  targetDirectory = file("${project.buildDir}/generated-sources/js2p")
  targetPackage = 'com.example'
  sourceFiles = [ 
                   [ 
                     path: 'path/to/file1',
                     classNamePrefix: 'Prefix'
                   ],
                   [ 
                     path: 'path/to/file2',
                     classNamePrefix: 'AnotherPrefix'
                   ],
                   [ 
                     path: 'path/to/file3',
                     classNamePrefix: 'ThirdPrefix',
                     usePrimitives: false      //Allow overiding the default
                   ]
                 ]
} 
```

**Comment by eirnym** [comment]
In this example I see `sourceFiles` as list of configuration objects. In such way we won't break object model of this configuration at the first place. Cutting down to a simple cases, such that we have no options to reconfigure for each path we have an example below. And within this example we have 2 major problems: 1) it's not a backward compatible way to define settings (I won't promise here to have no breaking changes in the future) 2) it's too complicated to understand and just write them down.

```gradle
jsonSchema2Pojo {

  //Default settings to be applied to all files unless otherwise specified
  generateBuilders = true
  usePrimitives = true
  targetDirectory = file("${project.buildDir}/generated-sources/js2p")
  targetPackage = 'com.example'
  sourceFiles = [ 
                   [ 
                     path: 'path/to/file1',
                   ],
                   [ 
                     path: 'path/to/file2',
                   ],
                   [ 
                     path: 'path/to/file3',
                   ]
                 ]
} 
```

To overcome the second issue I mentioned, we could add a `path` to main set. In this case we have a problematic situation as below, which doesn't describe what to do with third set (it obviously conflicts with "main" configuration.

 ```gradle
jsonSchema2Pojo {

  //Default settings to be applied to all files unless otherwise specified
  generateBuilders = true
  usePrimitives = true
  path: 'path/to/file'
  targetDirectory = file("${project.buildDir}/generated-sources/js2p")
  targetPackage = 'com.example'
  sourceFiles = [ 
                   [ 
                     path: 'path/to/file1',
                   ],
                   [ 
                     path: 'path/to/file2',
                   ],
                   [ 
                   ]
                 ]
} 
```

**Comment by eirnym** [comment]
BTW, I have a systematic question for such modules in projects. Why not to create a separate package for them, put them _once_ in a repository and forget about them before next change? I see that in Java on every corner, however, everything compiles much faster when you don't generate POJOs every time

**Comment by dan-lind** [comment]
> BTW, I have a systematic question for such modules in projects. Why not to create a separate package for them, put them _once_ in a repository and forget about them before next change? I see that in Java on every corner, however, everything compiles much faster when you don't generate POJOs every time

In my example above I just used 3 schemas, but in reality we have 12 schemas, and some of those are on revision 8 at this point. I'm very glad that we have the opportunity to regenerate the POJOs on change rather than having to manually update the models.

**Comment by eirnym** [comment]
Great, thank you

I think it'd be easy to implement. PRs are welcome

**Comment by eirnym** [comment]
@dan-lind Hey, sorry for the delay, I implemented the feature. Will add more tests for it later

**Comment by eirnym** [comment]
dsl is something like this:

```gradle
jsonSchema2Pojo {
   // parameters
  executions {
      executionName {
              // parameters
      }
  }
}
```

**Comment by eirnym** [comment]
@dan-lind 3.0.0 is just released with your feature in. After it'd be published, you're welcome to test!

**Comment by dan-lind** [comment]
Sure, I will take a look!
